nautobot/ipam/migrations/0031_ipam__prefix__add_parent.py

```diff
@@ -43,6 +45,64 @@ def reverse_it(apps, schema_editor):
         vrf.delete()
 
 
+def migrate_ipaddress_status_to_type(apps, schema_editor):
+    """Migrate IPAddress.status DHCP/SLAAC to types."""
+    Status = apps.get_model("extras", "Status")
+    IPAddress = apps.get_model("ipam", "IPAddress")
+    ContentType = apps.get_model("contenttypes", "ContentType")
+
+    ipaddress_ct = ContentType.objects.get_for_model(IPAddress)
+    statuses = Status.objects.filter(content_types=ipaddress_ct)
+    status_active = statuses.get(name="Active")
+
+    # Update all objects of status=DHCP to type=DHCP & status=Active.
+    print(">>> Migrating IPAddresses with status DHCP to type DHCP...")
+    IPAddress.objects.filter(status__name="DHCP").update(
+        type=nautobot.ipam.choices.IPAddressTypeChoices.TYPE_DHCP,
+        status=status_active,
+    )
+
+    # Update all objects of status=SLAAC to type=SLAAC & status=Active.
+    print(">>> Migrating IPAddresses with status SLAAC to type SLAAC...")
+    IPAddress.objects.filter(status__name="SLAAC").update(
+        type=nautobot.ipam.choices.IPAddressTypeChoices.TYPE_SLAAC,
+        status=status_active,
+    )
+
+    if IPAddress.objects.filter(models.Q(status__name="DHCP") | models.Q(status__name="SLAAC")).exists():
+        raise SystemExit("There should not be any IPAddress with type DHCP or SLAAC! This is bad!")
```

Comment:
Is this just checking that the above migrations didn't miss anything? If so we don't frequently post-check a batch update as simple as this. Regardless, let's improve the message so the user can know what to do if they encounter this.

Reply:
It was more for me, but yeah. Tests aren't quite fully passing yet.